app/models/labware_creators/multi_stamp_tubes.rb

```diff
@@ -78,6 +78,27 @@ def request_hash(transfer, child_plate)
       }
     end
 
+    def occupied_wells(wells)
+      wells.each_with_object([]) do |well, occupied_wells|
+        next if well.empty?
+
+        occupied_wells << well
+      end
+    end
+
+    def asset_groups(child_plate)
+      # split the wells by study id e.g. { '1': [<well1>, <well3>, <well4>], '2': [{<well2>, <well5>}]}
+      study_wells = occupied_wells(child_plate.wells).to_a.group_by { |well| well.aliquots.first.study.id }
```

Comment:
You've already have an array here, to `to_a` is redundant

Reply:
Also, it may be worth changing the plate lookup to include 'wells.aliquots.study' to improve performance significantly. 

I'd probably suggest `plate_with_custom_includes`. The individual named finders were based on the idea that we could encapsulate a lookup as its own distinct thing, but in practice things seem to diverge more than they share comonalities.

Reply:
Yep, done